packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -74,6 +80,16 @@ export namespace TerminalCommands {
         category: TERMINAL_CATEGORY,
         label: 'Create New Integrated Terminal'
     });
+    export const PROFILE_NEW = Command.toLocalizedCommand({
+        id: 'terminal:new:profile',
+        category: TERMINAL_CATEGORY,
+        label: 'Create New Integrated Terminal from a Profile'
+    });
+    export const PROFILE_DEFAULT = Command.toLocalizedCommand({
+        id: 'terminal:profile:default',
+        category: TERMINAL_CATEGORY,
+        label: 'Choose the default Terminal Profile'
+    });
```

Comment:
These commands do not seem to be present in VS Code, and are not added to the `terminal` main-menu.
I do see the command `'Create New Terminal (With Profile)'` however.

Reply:
terminal-frontend-contribution.ts L 653-668?

Reply:
I mean, these commands do not exist in VS Code, are they necessary?
Instead we should probably implement `Create New Terminal (With Profile)` which is present in VS Code and likely achieves what we are attempting here.

Reply:
Yes, the UI (in particular choosing the default profile) is very different from what VS Code does. As for the "Create New Terminal (With Profile)", I would argue that my solution is superior: the "..." at the end of "New Profile" is a well established UI pattern indicating that the user will face a dialog to do some configuration or selection before running the action.
Remember that "do the same as VS Code" is not a strategic goal of the project anymore.

Reply:
I don't really understand why "New Terminal" should not exist in VS Code: it exists and is under the "terminal" main menu. If you're saying we should use the same command ids, I'm not sure about that: are the command id's supposed to be API in VS Code?

Reply:
However in this case it adds entries in the `terminal` menu that downstream applications might not be expecting since in general we do follow the default implementation of VS Code. Moreover, plugins may attempt to trigger the `workbench.action.terminal.newWithProfile` command but it does not exist.

If we want to keep the commands that you've added on top of VS Code I'd maybe suggest we move them out of the `terminal` menu.

Reply:
> However in this case it adds entries in the terminal menu that downstream applications might not be expecting

Having the same menu structure as VS Code is not a promise I think we are making to our adopters. If downstream apps rely on that, we have a problem. Our UI with terminals and profiles is very different (no tabbed terminals, for example) and I feel it's a bit of a kludge in VS Code. It would be wrong to strive for visual compatibility in this respect, IMO.

> Moreover, plugins may attempt to trigger the workbench.action.terminal.newWithProfile command but it does not exist.

That's the question I'm asking: is this command id VS Code published API we should support?

Reply:
These are, though: https://code.visualstudio.com/api/references/commands

Reply:
> That's the question I'm asking: is this command id VS Code published API we should support? I don't think it is.

I believe it is the main reason that command ids follow VS Code, since they can be ultimately executed from plugins.
In the case that we do not match their id, we have a mapping performed in the plugin system, for example:
- https://github.com/eclipse-theia/theia/blob/d383fe25ee6dd2d0a3296222235b73e425666e2f/packages/plugin-ext-vscode/src/browser/plugin-vscode-commands-contribution.ts#L373-L375

I understand you might be referring to [builtin commands](https://code.visualstudio.com/api/references/commands), but the page also describes "common commands" and such commands are also accessible by plugins.

Reply:
Having a look at the broader picture here: all the other terminal-related commands do _not_ have the same command ids as in VS Code (e.g. `terminal:new` instead of `workbench.action.terminal.new`. So having just the two new one's aligned seems odd. In this PR, the command ids are consistent with the rest of the terminal commands. If we want to align the command ids with VS Code, it should be done in a separate PR that covers all terminal-related command ids.

Reply:
I've opened an issue: https://github.com/eclipse-theia/theia/issues/12084